automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Orchestration.scala

```diff
@@ -0,0 +1,185 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.config.{AuthToken, Config}
+
+import scala.concurrent.duration._
+
+
+trait Orchestration extends FireCloudClient with LazyLogging {
+
+  def responseAsList(response: String): List[Map[String, Object]] = {
+    mapper.readValue(response, classOf[List[Map[String, Object]]])
+  }
+
+  private def retry[T](remainingBackOffIntervals: Seq[FiniteDuration])(op: => () => Boolean): Boolean = {
+    op() match {
+      case true => true
+      case false if remainingBackOffIntervals.isEmpty => false
+      case false if remainingBackOffIntervals.nonEmpty =>
+        Thread sleep remainingBackOffIntervals.head.toMillis
+        retry(remainingBackOffIntervals.tail)(op)
+    }
+  }
+
+  object billing {
+
+    object BillingProjectRole extends Enumeration {
```

Comment:
Is that 2.12 only? I think we're on 2.11. I read up on enumeration patterns before I chose this one. Should be try to be on 2.12 if it would be easy to move?